python/lsst/daf/butler/script/removeRuns.py

```diff
@@ -113,10 +124,15 @@ def removeRuns(
     """
     runs, datasets = _getCollectionInfo(repo, collection)
 
-    def doRemove(runs: Sequence[str]) -> None:
+    def doRemove(runs: Sequence[RemoveRun]) -> None:
         """Perform the remove step."""
         butler = Butler(repo, writeable=True)
-        butler.removeRuns(runs, unstore=True)
+        for run in runs:
```

Comment:
Should this entire loop be in a transaction in case something goes wrong before the removeRuns happens and you may not want some of the chains to have been modified and some unchanged?

Reply:
that seems like a good idea. I think to do that I put the loop inside a block that begins with `with butler.transaction():`, and that's all, is that right?